applications/pelion_client/README.rst

```diff
@@ -0,0 +1,516 @@
+| ``oma_stopwatch``     |                               | Registers the respective (Open Mobile Alliance) object resources, which work as an example of the device communication with the cloud. |
+|                       |                               | This resource counts time, increasing it periodically.                                                                                 |
+|                       |                               | You can configure the period and reset the time count.                                                                                 |
++-----------------------+-------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
+| ``led_state``         |                               | Reflects the connection state onto LEDs.                                                                                               |
++-----------------------+-------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
+| ``power_manager``     |                               | Switches the device into the low power mode when no interaction is detected for a given period of time.                                |
++-----------------------+-------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
+| ``dfu``               |                               | Confirms that the DFU image is working fine after the update process.                                                                  |
++-----------------------+-------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
+
+
+On top of these modules, the application uses the following :ref:`lib_caf` modules:
+
+* :ref:`caf_buttons`
+* :ref:`caf_leds`
+
+The following figure shows the application architecture.
+The figure visualizes relations between Event Manager, modules, drivers, and libraries.
+
+.. figure:: /images/pelion_client_architecture.svg
+    :alt: Module hierarchy
+
+    Relationship between modules and the event manager
+
+.. _pelion_client_memory_allocation:
+
+Memory allocation
+=================
+
+The application allocates memory dynamically for the following features:
+
+* Event manager events - Memory for these events is allocated from Zephyr's :ref:`System Heap <zephyr:heap_v2>`.
+* Pelion library - Memory for this library is allocated from the libc heap.
+* mbed TLS - Memory for this library is allocated from either the libc heap or a dedicated buffer.
```

Comment:
I'm going to stick to `mbed TLS`, because of how we write it in our docs. The source of https://tls.mbed.org/ is inconsistent and uses both `Mbed` and `mbed`.